rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -137,6 +140,23 @@ pub struct StreamProcessor<TPayload: Clone> {
     dlq_policy: Option<DlqPolicy<TPayload>>,
 }
 
+impl StreamProcessor<KafkaPayload> {
+    pub fn with_kafka<F: ProcessingStrategyFactory<KafkaPayload> + 'static>(
+        config: KafkaConfig,
+        factory: F,
+        topic: Topic,
+        dlq_policy: Option<DlqPolicy<KafkaPayload>>,
+    ) -> Self {
+        let consumer_state = Arc::new(Mutex::new(ConsumerState::new(Box::new(factory))));
+        let callbacks = Callbacks(consumer_state.clone());
+
+        // TODO: Can this fail?
```

Comment:
I guess so, but then there's something wrong either network or broker related, so letting the consumer crash and come back up and retry might just be the safest thing to do in this scenario.